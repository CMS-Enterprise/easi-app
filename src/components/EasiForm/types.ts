import React from 'react';
import {
  FieldPath,
  FieldValues,
  RefCallBack,
  RegisterOptions,
  UseFormRegisterReturn,
  UseFormReturn
} from 'react-hook-form';

export type EasiPartialSubmitProps<TFieldValues extends FieldValues> = {
  update: (
    /** Object containing valid dirty field values */
    formData: Partial<TFieldValues>
  ) => Promise<any>;
  /** Callback to execute after updating */
  callback?: () => void;
  /** Whether to clear field error messages */
  clearErrors?: boolean;
};

export type EasiPartialSubmit<TFieldValues extends FieldValues> = ({
  update,
  clearErrors,
  callback
}: EasiPartialSubmitProps<TFieldValues>) => Promise<void>;

export type EasiFormRegister<TFieldValues extends FieldValues> = <
  TFieldName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>
>(
  name: TFieldName,
  options?: RegisterOptions<TFieldValues, TFieldName>
) => UseFormRegisterReturn<TFieldName> & { inputRef: RefCallBack };

export type UseEasiFormReturn<
  TFieldValues extends FieldValues = FieldValues,
  TContext = any
> = Omit<UseFormReturn<TFieldValues, TContext>, 'register'> & {
  /** Ignore errors and submit valid dirty field values */
  partialSubmit: EasiPartialSubmit<TFieldValues>;
  /** Custom register function that returns `inputRef` for function components */
  register: EasiFormRegister<TFieldValues>;
};

export type EasiFormProviderProps<
  TFieldValues extends FieldValues,
  TContext = any
> = {
  children: React.ReactNode | React.ReactNode[];
} & UseEasiFormReturn<TFieldValues, TContext>;
