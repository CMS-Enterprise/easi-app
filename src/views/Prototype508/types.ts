import { DateTime } from 'luxon';

export enum DocumentType {
  TestPlan,
  TestingVPAT,
  TestResults,
  AwardedVPAT,
  RemediationPlan
}

export type Document = {
  type: DocumentType;
  mimetype: string;
  createdAt: DateTime;
};

export enum ProjectStatus {
  ConsultRequested,
  TestingRequested,
  TestingInProgress,
  TestingCompleted,
  InRemediation
}

export type BusinessOwner = {
  name: string;
};

export enum ActivityType {
  NoteAdded,
  StatusChanged,
  DocumentAdded,
  DocumentRemoved,
  ProjectCreated
}

export type Activity = {
  id: number;
  authorName: string;
  content: string;
  createdAt: DateTime;
  type: ActivityType;
};

export type Project = {
  id: number;
  name: string;
  status: ProjectStatus;
  businessOwner: BusinessOwner;
  submissionDate: DateTime;
  lastUpdatedAt: DateTime;
  lifecycleID: string;
  description?: string;
  documents: Document[];
  activities: Activity[];
};
