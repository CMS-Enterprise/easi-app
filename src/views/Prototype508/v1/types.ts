import { DateTime } from 'luxon';

export enum DocumentType {
  TestPlan = 'Testing Plan',
  TestingVPAT = 'Testing VPAT',
  TestResults = 'Test Results',
  AwardedVPAT = 'Awarded VPAT',
  RemediationPlan = 'Remediation Plan'
}

export type Document = {
  id: number;
  type: DocumentType;
  mimetype: string;
  createdAt: DateTime;
  score?: number;
};

export enum ProjectStatus {
  ConsultRequested = 'Consult Requested',
  TestingRequested = 'Testing Requested',
  TestingInProgress = 'Testing in Progress',
  TestingCompleted = 'Testing Completed',
  InRemediation = 'In Remediation'
}

export type BusinessOwner = {
  name: string;
};

export enum ActivityType {
  NoteAdded,
  StatusChanged,
  DocumentAdded,
  DocumentRemoved,
  ProjectCreated
}

export type Activity = {
  id: number;
  authorName: string;
  content: string;
  createdAt: DateTime;
  type: ActivityType;
};

export type Project = {
  id: number;
  name: string;
  status: ProjectStatus;
  businessOwner: BusinessOwner;
  submissionDate: DateTime;
  lastUpdatedAt: DateTime;
  lifecycleID: string;
  description?: string;
  documents: Document[];
  activities: Activity[];
  banner?: string;
};
