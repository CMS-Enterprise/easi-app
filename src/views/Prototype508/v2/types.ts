import { DateTime } from 'luxon';

import { ProgressStatus } from './components/Progress';

export enum DocumentType {
  TestPlan = 'Testing Plan',
  TestingVPAT = 'Testing VPAT',
  TestResults = 'Test Results',
  AwardedVPAT = 'Awarded VPAT',
  RemediationPlan = 'Remediation Plan',
  Other = 'Other Document'
}

export type Document = {
  id: number;
  type: DocumentType;
  otherName?: string;
  mimetype: string;
  createdAt: DateTime;
  testDate?: DateTime;
  score?: number;
};

export type Person = {
  name: string;
  component: string;
};

export type Note = {
  id: number;
  authorName: string;
  content: string;
  createdAt: DateTime;
};

export enum RequestStep {
  RequestReceived = 'Request Received',
  DocumentsReceived = 'Documents Received',
  TestScheduled = 'Test Scheduled',
  RemediationRequested = 'Remediation Requested',
  RemediationInProgress = 'Remediation in Progress',
  ValidationTestingScheduled = 'Validation Testing Scheduled',
  Completed = 'Completed'
}

export type RequestStepStatus = {
  date: DateTime;
  status: ProgressStatus;
};

export type PastRequest = {
  name: string;
  release: string;
  lastTestDate: DateTime;
};

export type Project = {
  id: number;
  name: string;
  release?: string;
  status: RequestStep;
  businessOwner: Person;
  pointOfContact: Person;
  testDate?: DateTime;
  submissionDate: DateTime;
  remediationStartDate?: DateTime;
  lastUpdatedAt: DateTime;
  lifecycleID: string;
  description?: string;
  documents: Document[];
  notes: Note[];
  banner?: string;
  stepStatuses: { [index in RequestStep]?: RequestStepStatus };
  pastRequests: PastRequest[];
};
