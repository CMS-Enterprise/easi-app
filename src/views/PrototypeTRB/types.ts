import { DateTime } from 'luxon';

export enum DocumentType {
  FindingsLetter = 'Findings Letter',
  ArchitectureDiagram = 'Architecture Diagram',
  Deck = 'Deck'
}

export type Document = {
  id: number;
  type: DocumentType;
  mimetype: string;
  createdAt: DateTime;
  score?: number;
};

export enum ProjectStatus {
  ReviewRequested = 'Review Requested',
  InConsultation = 'In Consultation',
  InDesignReview = 'In Design Review',
  InOperationalReadiness = 'In Operational Readiness',
  WaitingForFindings = 'Waiting for Findings',
  Closed = 'Closed'
}

export type BusinessOwner = {
  name: string;
};

export enum ActivityType {
  NoteAdded,
  StatusChanged,
  DocumentAdded,
  DocumentRemoved,
  ProjectCreated
}

export type Activity = {
  id: number;
  authorName: string;
  content: string;
  createdAt: DateTime;
  type: ActivityType;
};

export type Project = {
  id: number;
  name: string;
  status: ProjectStatus;
  businessOwner: BusinessOwner;
  submissionDate: DateTime;
  lastUpdatedAt: DateTime;
  lifecycleID: string;
  description?: string;
  documents: Document[];
  activities: Activity[];
  banner?: string;
};
