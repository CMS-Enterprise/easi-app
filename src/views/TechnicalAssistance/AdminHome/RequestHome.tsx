import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { CardGroup, IconArrowForward, Link } from '@trussworks/react-uswds';
import { DateTime } from 'luxon';

import UswdsReactLink from 'components/LinkWrapper';
import PageLoading from 'components/PageLoading';
import Alert from 'components/shared/Alert';
import Divider from 'components/shared/Divider';
import InitialsIcon from 'components/shared/InitialsIcon';
import useCacheQuery from 'hooks/useCacheQuery';
import GetTrbRequestHomeQuery from 'queries/GetTrbRequestHomeQuery';
import {
  GetTrbRequestHome as GetTrbRequestHomeType,
  GetTrbRequestHomeVariables
} from 'queries/types/GetTrbRequestHome';
import { TRBRequestStatus } from 'types/graphql-global-types';
import { TrbAdminPageProps } from 'types/technicalAssistance';
import { formatDateLocal } from 'utils/date';
import getPersonNameAndComponentVal from 'utils/getPersonNameAndComponentVal';

import InformationCard from './components/InformationCard';
import TrbAdminWrapper from './components/TrbAdminWrapper';
import { TrbAssignLeadModalOpener } from './TrbAssignLeadModal';

const RequestHome = ({
  trbRequest,
  assignLeadModalRef,
  assignLeadModalTrbRequestIdRef
}: TrbAdminPageProps) => {
  const { t } = useTranslation('technicalAssistance');

  const { id } = trbRequest;

  const { data, loading } = useCacheQuery<
    GetTrbRequestHomeType,
    GetTrbRequestHomeVariables
  >(GetTrbRequestHomeQuery, {
    variables: { id }
  });

  const { consultMeetingTime, trbLeadInfo, documents } = data?.trbRequest || {};

  if (loading) return <PageLoading />;

  return (
    <TrbAdminWrapper
      activePage="request"
      trbRequestId={id}
      title={t('adminHome.requestHome')}
      noteCount={trbRequest.adminNotes.length}
      adminActionProps={{
        status: trbRequest.status,
        state: trbRequest.state,
        assignLeadModalTrbRequestIdRef,
        assignLeadModalRef
      }}
    >
      {data?.trbRequest && (
        /* Consult details */
        <>
          <h2 className="margin-top-4 margin-bottom-3">
            {t('adminHome.consultDetails')}
          </h2>

          <p className="text-bold margin-bottom-1">{t('adminHome.dateTime')}</p>

          {trbRequest.status === TRBRequestStatus.REQUEST_FORM_COMPLETE ||
          trbRequest.status === TRBRequestStatus.DRAFT_REQUEST_FORM ||
          trbRequest.status === TRBRequestStatus.NEW ? (
            // Unable to set consult until initial request form complete
            <Alert type="info" slim className="margin-y-1 margin-top-2">
              {t('adminHome.reviewInitialRequest')}
            </Alert>
          ) : (
            <div>
              {!consultMeetingTime ? (
                // Set meeting time
                <UswdsReactLink
                  className="usa-button usa-button--outline margin-top-1"
                  variant="unstyled"
                  to="request/schedule-consult"
                >
                  {t('adminHome.addDateTime')}
                </UswdsReactLink>
              ) : (
                // Display meeting time
                <div className="display-flex flex-align-center">
                  <p className="margin-right-2 margin-y-0">
                    {t('adminHome.consultDate', {
                      date: formatDateLocal(consultMeetingTime, 'MM/dd/yyyy'),
                      time: DateTime.fromISO(consultMeetingTime).toLocaleString(
                        DateTime.TIME_SIMPLE
                      )
                    })}
                  </p>
                  <UswdsReactLink to="request/schedule-consult">
                    {t('adminHome.change')}
                  </UswdsReactLink>
                </div>
              )}
            </div>
          )}

          {/* TRB lead details */}
          <h3 className="margin-top-4 margin-bottom-3">
            {t('adminHome.representative')}
          </h3>

          <p className="text-bold">{t('adminHome.trbLead')}</p>

          {trbLeadInfo?.commonName ? (
            <>
              <div className="display-flex flex-align-center">
                <InitialsIcon
                  name={trbLeadInfo.commonName}
                  index={0}
                  className="margin-right-1"
                />
                <p className="text-bold margin-0">
                  {getPersonNameAndComponentVal(trbLeadInfo.commonName, 'TRB')}
                </p>
              </div>
              <Link
                aria-label={t('adminHome.sendEmail')}
                className="line-height-body-5 margin-left-5"
                href={trbLeadInfo.email}
                target="_blank"
              >
                {trbLeadInfo.email}
              </Link>
            </>
          ) : (
            <TrbAssignLeadModalOpener
              trbRequestId={trbRequest.id}
              modalRef={assignLeadModalRef}
              trbRequestIdRef={assignLeadModalTrbRequestIdRef}
              className="usa-button--outline"
            >
              {t('adminHome.assignLead')}
            </TrbAssignLeadModalOpener>
          )}

          <Divider className="margin-top-6 margin-bottom-5" />

          {/* Forms and Documents */}
          <h2 className="margin-y-3">{t('adminHome.formAndDocs')}</h2>

          <CardGroup className="tablet:grid-col-10">
            <InformationCard
              type="initialRequestForm"
              trbRequest={data?.trbRequest}
            />

            <InformationCard
              type="adviceLetter"
              trbRequest={data?.trbRequest}
            />
          </CardGroup>

          <p className="text-bold margin-bottom-105">
            {t('adminHome.supportingDocs')}
          </p>

          {/* Documents count */}
          <div data-testid="document-count">
            <Trans
              i18nKey={
                documents?.length === 1
                  ? 'technicalAssistance:adminHome.docInfo'
                  : 'technicalAssistance:adminHome.docInfoPlural'
              }
              components={{
                bold: <span className="text-bold" />,
                docCount: documents?.length || 0
              }}
            />
          </div>

          <UswdsReactLink
            to="documents"
            className="display-flex flex-align-center margin-top-2"
          >
            {t('adminHome.viewDocs')}
            <IconArrowForward className="margin-left-1" />
          </UswdsReactLink>
        </>
      )}
    </TrbAdminWrapper>
  );
};

export default RequestHome;
