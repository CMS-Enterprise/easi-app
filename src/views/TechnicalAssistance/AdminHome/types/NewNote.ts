/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { TRBAdminNoteCategory } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL fragment: NewNote
// ====================================================

export interface NewNote_author {
  __typename: "UserInfo";
  commonName: string;
}

export interface NewNote {
  __typename: "TRBAdminNote";
  id: UUID;
  isArchived: boolean;
  category: TRBAdminNoteCategory;
  noteText: string;
  author: NewNote_author;
  createdAt: Time;
}
