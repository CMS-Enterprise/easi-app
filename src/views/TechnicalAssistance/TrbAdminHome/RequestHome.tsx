import React from 'react';
import { useTranslation } from 'react-i18next';

import { TrbAdminPage } from 'types/technicalAssistance';

const RequestHome: TrbAdminPage = trbRequestId => {
  const { t } = useTranslation('technicalAssistance');
  return (
    <div
      className="trb-admin-home__request-home"
      data-testid="trb-admin-home__request-home"
      id={`trbAdminRequestHome-${trbRequestId}`}
    >
      <h1 className="margin-y-0">{t('adminHome.subnav.requestHome')}</h1>
    </div>
  );
};

export default RequestHome;
