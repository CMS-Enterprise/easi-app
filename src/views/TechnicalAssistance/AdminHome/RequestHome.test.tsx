import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { render, waitForElementToBeRemoved } from '@testing-library/react';
import { ModalRef } from '@trussworks/react-uswds';
import i18next from 'i18next';

import { trbRequestSummary } from 'data/mock/trbRequest';
import GetTrbRequestHomeQuery from 'queries/GetTrbRequestHomeQuery';
import {
  TRBAdviceLetterStatus,
  TRBFormStatus,
  TRBRequestStatus
} from 'types/graphql-global-types';
import { TrbRequestIdRef } from 'types/technicalAssistance';
import VerboseMockedProvider from 'utils/testing/VerboseMockedProvider';

import RequestHome from './RequestHome';

describe('Trb Admin Request Home', () => {
  Element.prototype.scrollIntoView = jest.fn();

  const trbRequestId = trbRequestSummary.id;
  const modalRef = React.createRef<ModalRef>();
  const trbRequestIdRef = React.createRef<TrbRequestIdRef>();

  it('renders successfully with empty data', async () => {
    const { getByText, asFragment, getByRole, getByTestId } = render(
      <VerboseMockedProvider
        defaultOptions={{
          watchQuery: { fetchPolicy: 'no-cache' },
          query: { fetchPolicy: 'no-cache' }
        }}
        mocks={[
          {
            request: {
              query: GetTrbRequestHomeQuery,
              variables: {
                id: trbRequestId
              }
            },
            result: {
              data: {
                trbRequest: {
                  id: trbRequestId,
                  consultMeetingTime: null,
                  taskStatuses: {
                    formStatus: TRBFormStatus.READY_TO_START,
                    adviceLetterStatus: TRBAdviceLetterStatus.CANNOT_START_YET,
                    __typename: 'TRBTaskStatuses'
                  },
                  form: {
                    id: 'c92ec6a6-cd5b-4be3-895a-e88f7de76c22',
                    modifiedAt: null,
                    __typename: 'TRBRequestForm'
                  },
                  adviceLetter: null,
                  trbLeadInfo: {
                    commonName: '',
                    email: '',
                    __typename: 'UserInfo'
                  },
                  documents: [],
                  adminNotes: [],
                  __typename: 'TRBRequest'
                }
              }
            }
          }
        ]}
      >
        <MemoryRouter initialEntries={[`/trb/${trbRequestId}/request`]}>
          <Route exact path="/trb/:id/:activePage">
            <RequestHome
              trbRequestId={trbRequestId}
              trbRequest={trbRequestSummary}
              assignLeadModalRef={modalRef}
              assignLeadModalTrbRequestIdRef={trbRequestIdRef}
            />
          </Route>
        </MemoryRouter>
      </VerboseMockedProvider>
    );

    await waitForElementToBeRemoved(() => getByTestId('page-loading'));

    expect(
      getByText(i18next.t<string>('technicalAssistance:adminHome.requestHome'))
    ).toBeInTheDocument();

    // Consult meeting time
    expect(
      getByText(
        i18next.t<string>('technicalAssistance:adminHome.reviewInitialRequest')
      )
    ).toBeInTheDocument();

    // Assign lead button
    expect(
      getByRole('button', {
        name: i18next.t<string>('technicalAssistance:adminHome.assignLead')
      })
    ).toBeInTheDocument();

    // Document count
    expect(getByTestId('document-count').textContent).toBe(
      'There are 0 additional documents uploaded as a part of this request.'
    );

    // Start advice letter button disabled
    expect(
      getByRole('button', {
        name: i18next.t<string>('technicalAssistance:adminHome.startAdvice')
      })
    ).toBeDisabled();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders successfully with populated data', async () => {
    const { getByText, getByRole, getByTestId } = render(
      <VerboseMockedProvider
        defaultOptions={{
          watchQuery: { fetchPolicy: 'no-cache' },
          query: { fetchPolicy: 'no-cache' }
        }}
        mocks={[
          {
            request: {
              query: GetTrbRequestHomeQuery,
              variables: {
                id: trbRequestId
              }
            },
            result: {
              data: {
                trbRequest: {
                  id: trbRequestId,
                  consultMeetingTime: '2024-01-05T05:00:00Z',
                  taskStatuses: {
                    formStatus: TRBFormStatus.COMPLETED,
                    adviceLetterStatus: TRBAdviceLetterStatus.READY_TO_START,
                    __typename: 'TRBTaskStatuses'
                  },
                  form: {
                    id: 'c92ec6a6-cd5b-4be3-895a-e88f7de76c22',
                    modifiedAt: '2023-01-05T05:00:00Z',
                    __typename: 'TRBRequestForm'
                  },
                  adviceLetter: {
                    __typename: 'TRBAdviceLetter',
                    id: '123',
                    modifiedAt: '2023-02-05T05:00:00Z'
                  },
                  trbLeadInfo: {
                    commonName: 'Jerry Seinfeld',
                    email: 'js@oddball.io',
                    __typename: 'UserInfo'
                  },
                  documents: [
                    {
                      __typename: 'TRBRequestDocument',
                      id: '456'
                    }
                  ],
                  adminNotes: [],
                  __typename: 'TRBRequest'
                }
              }
            }
          }
        ]}
      >
        <MemoryRouter initialEntries={[`/trb/${trbRequestId}/request`]}>
          <Route exact path="/trb/:id/:activePage">
            <RequestHome
              trbRequestId={trbRequestId}
              trbRequest={{
                ...trbRequestSummary,
                status: TRBRequestStatus.CONSULT_SCHEDULED
              }}
              assignLeadModalRef={modalRef}
              assignLeadModalTrbRequestIdRef={trbRequestIdRef}
            />
          </Route>
        </MemoryRouter>
      </VerboseMockedProvider>
    );

    await waitForElementToBeRemoved(() => getByTestId('page-loading'));

    // Consult meeting time
    expect(getByText('01/05/2024 at 5:00 AM')).toBeInTheDocument();

    // TRB lead info
    expect(getByText('Jerry Seinfeld, TRB')).toBeInTheDocument();

    expect(getByTestId('document-count').textContent).toBe(
      'There is 1 additional document uploaded as a part of this request.'
    );

    expect(
      getByRole('button', {
        name: i18next.t<string>('technicalAssistance:adminHome.startAdvice')
      })
    ).not.toBeDisabled();
  });
});
