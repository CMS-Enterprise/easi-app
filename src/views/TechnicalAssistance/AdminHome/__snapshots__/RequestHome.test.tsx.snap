// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trb Admin Request Home renders successfully with empty data 1`] = `
<DocumentFragment>
  <div
    class="trbAdmin__request margin-bottom-2"
    data-testid="trb-admin-home__request"
    id="trbAdmin__request"
  >
    <div
      class="grid-row grid-gap-lg"
      data-testid="grid"
    >
      <div
        class="tablet:grid-col-8"
        data-testid="grid"
      >
        <h1
          class="margin-top-0 margin-bottom-1"
        >
          Request home
        </h1>
      </div>
      <div
        class="tablet:grid-col-4"
        data-testid="grid"
      >
        <div
          class="trb-admin-home__request-notes bg-base-lightest padding-x-2 padding-y-1 margin-bottom-3"
          data-testid="trb-admin-home__request-notes"
        >
          <p
            class="margin-y-0 line-height-body-5"
            data-testid="note-box-count"
          >
            <span
              class="text-bold"
            >
              1
            </span>
             note about this request
          </p>
          <div
            class="margin-top-05 display-flex"
          >
            <div
              class="desktop:grid-col-6"
              data-testid="grid"
            >
              <button
                class="usa-button usa-button--unstyled"
                data-testid="note-modal-button"
                type="button"
              >
                View notes
              </button>
            </div>
            <div
              class="desktop:grid-col-6"
              data-testid="grid"
            >
              <button
                class="usa-button usa-button--unstyled"
                data-testid="note-modal-button"
                type="button"
              >
                Add a note
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <h2
      class="margin-top-4 margin-bottom-3"
    >
      Consult meeting details
    </h2>
    <p
      class="text-bold margin-bottom-1"
    >
      Date and time
    </p>
    <div
      class="usa-alert usa-alert--info usa-alert--slim margin-y-1 margin-top-2"
      data-testid="alert"
    >
      <div
        class="usa-alert__body"
      >
        <p
          class="usa-alert__text"
        >
          Please wait until the initial request form is submitted, then review the form before setting a date and time in EASi.
        </p>
      </div>
    </div>
    <h3
      class="margin-top-4 margin-bottom-3"
    >
      TRB representatives
    </h3>
    <p
      class="text-bold"
    >
      TRB Lead
    </p>
    <button
      class="usa-button usa-button--outline"
      type="button"
    >
      Assign a TRB Lead
    </button>
    <div
      class="easi-divider margin-top-6 margin-bottom-5"
    />
    <h2
      class="margin-y-3"
    >
      Forms and documents
    </h2>
    <ul
      class="usa-card-group tablet:grid-col-10"
      data-testid="CardGroup"
    >
      <li
        class="usa-card flex-1 information-card margin-bottom-2"
        data-testid="Card"
      >
        <div
          class="usa-card__container"
        >
          <header
            class="usa-card__header"
            data-testid="CardHeader"
          >
            <h3
              class="margin-0"
            >
              Initial request form
            </h3>
            <p
              class="text-base margin-0"
            >
              Completed by requester
            </p>
          </header>
          <div
            class="usa-card__body"
            data-testid="CardBody"
          >
            <div
              class="display-flex flex-align-center line-height-body-5 margin-bottom-2"
            >
              <span
                class="easi-tag easi-task-status-tag bg-info-light"
                data-testid="task-list-task-tag"
              >
                Ready to start
              </span>
            </div>
            <dt
              class="text-bold"
            >
              Last updated
            </dt>
            <dd
              class="margin-left-0 margin-bottom-1"
            >
              Not started
            </dd>
          </div>
          <div
            class="usa-card__footer"
            data-testid="CardFooter"
          >
            <button
              class="usa-button usa-button--outline"
              data-testid="button"
              type="button"
            >
              View
            </button>
          </div>
        </div>
      </li>
      <li
        class="usa-card flex-1 information-card margin-bottom-2"
        data-testid="Card"
      >
        <div
          class="usa-card__container"
        >
          <header
            class="usa-card__header"
            data-testid="CardHeader"
          >
            <h3
              class="margin-0"
            >
              Advice letter
            </h3>
            <p
              class="text-base margin-0"
            >
              To be completed by TRB
            </p>
          </header>
          <div
            class="usa-card__body"
            data-testid="CardBody"
          >
            <div
              class="display-flex flex-align-center line-height-body-5 margin-bottom-2"
            >
              <span
                class="easi-tag easi-task-status-tag border-2px text-base"
                data-testid="task-list-task-tag"
              >
                Cannot start yet
              </span>
            </div>
            <dt
              class="text-bold"
            >
              Last updated
            </dt>
            <dd
              class="margin-left-0 margin-bottom-1"
            >
              Not started
            </dd>
          </div>
          <div
            class="usa-card__footer"
            data-testid="CardFooter"
          >
            <button
              class="usa-button"
              data-testid="button"
              disabled=""
              type="button"
            >
              Start advice letter
            </button>
          </div>
        </div>
      </li>
    </ul>
    <p
      class="text-bold margin-bottom-105"
    >
      Supporting documents
    </p>
    <div
      data-testid="document-count"
    >
      There are 
      <span
        class="text-bold"
      >
        0
      </span>
       additional documents uploaded as a part of this request.
    </div>
    <a
      class="usa-link display-flex flex-align-center margin-top-2"
      href="/trb/d8ebedca-0031-4ccd-9690-37390726c50e/documents"
    >
      View documents
      <svg
        class="usa-icon margin-left-1"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
        />
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
